src/python/zquantum/core/graph.py

```diff
@@ -97,11 +95,103 @@ def generate_random_graph_erdos_renyi(
 
     # iterate through all pairs of nodes
     for pair in combinations(range(0, num_nodes), 2):
-        if uniform(0, 1) < probability:  # with the given probability
+        if uniform(0,1) < probability: # with the given probability
             if random_weights:
                 weight = uniform(0, 1)
             else:
                 weight = 1.0
             output_graph.add_edge(pair[0], pair[1], weight=weight)
+    
+    return output_graph
 
+def generate_random_graph_regular(num_nodes: int, 
+                                    degree: int, 
+                                    random_weights: bool = False) -> nx.Graph:
+    """Randomly generate a d-regular graph. 
+    A graph is generated by picking uniformly a graph among the set of graphs 
+    with the desired number of nodes and degre.
+
+    Args:
+        num_nodes: integer
+            Number of nodes.
+        degre: int
+            Degre of each edge.
+        random_weights: bool
+            Flag indicating whether the weights should be random or constant.
+    
+    Returns:
+        A networkx.Graph object
+    """
+    output_graph = nx.random_regular_graph(d = degree, n = num_nodes)
+    output_graph = weight_graph_edges(output_graph, random_weights)
+
+    return output_graph
+
+def generate_complete_graph(num_nodes: int, 
```

Comment:
Actually, we don't need this, because you can create Erdos-Renyi graph with edge probability equal to 1 and you'll get the same result.